currentstateserver/storage/postgres/current_room_state_table.go

```diff
@@ -81,6 +82,11 @@ const selectJoinedUsersSetForRoomsSQL = "" +
 	"SELECT state_key, COUNT(room_id) FROM currentstate_current_room_state WHERE room_id = ANY($1) AND" +
 	" type = 'm.room.member' and content_value = 'join' GROUP BY state_key"
 
+const selectKnownUsersSQL = "" +
+	"SELECT DISTINCT state_key FROM currentstate_current_room_state WHERE room_id = ANY(" +
+	"  SELECT DISTINCT room_id FROM currentstate_current_room_state WHERE state_key=$1 AND TYPE='m.room.member' AND content_value='join'" +
+	") AND TYPE='m.room.member' AND content_value='join' AND state_key LIKE $2 LIMIT $3"
```

Comment:
Comment explaining the subselect would be good. I _think_ it's so we only query joined users for rooms that the user is in?

Reply:
I've added comments. Yep, the goal is exactly that - that we return users based on the rooms that the searching user is joined to.